lib/elastomer/client/scroller.rb

```diff
@@ -18,12 +19,27 @@ class Client
     #     document['_source']
     #   end
     #
-    # Returns a new Scan instance
+    # Returns a new Scroller instance
     def scan( query, opts = {} )
-      Scan.new self, query, opts
+      opts = opts.merge(:search_type => 'scan')
+      Scroller.new(self, query, opts)
     end
 
-    # Continue scrolling a scan query.
+    #
+    #
+    def scroll( query, opts = {} )
+      Scroller.new(self, query, opts)
+    end
+
+    #
+    #
+    def start_scroll( opts = {} )
+      opts = opts.merge :action => 'search.start_scroll'
+      response = get '{/index}{/type}/_search', opts
+      response.body
+    end
```

Comment:
When you initiate a scroll, you can specify the index and the document type the query will execute against. The initial call returns the `scroll_id`. To continue scrolling you pass this `scroll_id` to the `/_search/scroll` API endpoint. The index and document type are no longer needed since that information is maintained by ES internally and retrieved via the `scroll_id`.

What is now the `start_scroll` method was originally embedded in the `Scan` class as part of the first `scroll_id` method call. It was moved out into its own method for (a) clarity and (b) because the initial results returned by a `scan` query do _not_ contain any document hits. But the initial call with a regular `scroll` query does include document hits. Take a look at the `Scroller#do_scroll` method for the gory details.